### PR TITLE
Add support for CMake config and find_package

### PR DESCRIPTION
**Pull request by AurelienBallier**
Hi,

Just some cmake update to create proper CMake files and add support for find_package in other packages.

Regards,

### CODE REVIEW

**Comment by Stinkfist0** [comment]
ping @juj ;)

File: MathGeoLibConfigVersion.cmake.in [hunk 1]
```diff
@@ -0,0 +1,11 @@
+set(PACKAGE_VERSION "@FOOBAR_VERSION@")
```
Comment (juj) [inline MathGeoLibConfigVersion.cmake.in]: What is `@FOOBAR_VERSION@`? This is perhaps leftover typo, and was sourced from another CMake script?

**Comment by juj** [comment]
> ping @juj ;)

Thanks, and sorry, this had fallen through my email queue. Looks good to me, though one question about the script above.

**Comment by AurelienBallier** [comment]
Yes you were right about FOOBAR_VERSION, version in generated MathGeoLibConfigVersion.cmake was not filled properly.
I also added build/ to .gitignore to avoid pushing build files to git like I just did.

**Comment by juj** [comment]
Great!

**Pull request merged**